File: lib/java/src/main/java/io/github/brenbar/japi/_UFieldDeclaration.java
package io.github.brenbar.japi;

public class _UFieldDeclaration {
    public final String fieldName;
    public final _UTypeDeclaration typeDeclaration;
    public final boolean optional;

    public _UFieldDeclaration(
            String fieldName,
            _UTypeDeclaration typeDeclaration,
            boolean optional) {
        this.fieldName = fieldName;
        this.typeDeclaration = typeDeclaration;
        this.optional = optional;
    }
}

File: tool/uapicodegen/uapicodegen/uapi/internal/types/UFieldDeclaration.py
from typing import Optional, TYPE_CHECKING

if TYPE_CHECKING:
    from ...internal.types.UTypeDeclaration import UTypeDeclaration


class UFieldDeclaration:
    def __init__(
            self,
            field_name: str,
            type_declaration: 'UTypeDeclaration',
            optional: bool
    ) -> None:
        self.field_name = field_name
        self.type_declaration = type_declaration
        self.optional = optional
